tests/main/snap-confine-privs/task.yaml

```diff
@@ -0,0 +1,60 @@
+summary: ensure that snap-confine handles group and user privileges correctly
+details: |
+    The openSUSE security team has made a remark about a particular part of
+    snap-confine's UID/GID handling. The code there was, we believe, correct
+    but this test is here to demonstrate that and ensure it never regresses.
+
+    Security review https://bugzilla.opensuse.org/show_bug.cgi?id=986050
+# This test is not executed on a core system simply because of the hassle of
+# building the support C program. In the future it might be improved with the
+# use of the classic snap where we just use classic to build the helper.
```

Comment:
Note, core has python3 and python3 has os.getresuid(). I think it is fine to test on just classic, but if you really want it everywhere, keep that in mind. Also, I think this test is valid on other distros where snap-confine is setuid. On those with fscaps, we'd of course need different tests.